### PR TITLE
Make the attention mask a mandatory argument of models

### PR DESCRIPTION
**Pull request by danieldk**
## Description

It is very easy to forget to pass the attention mask, giving bad output.
On the other hand it is very unlikely that someone doesn't want to use
an attention mask. In such cases, it is always possible to create a
custom mask all elements set to `True`.

While at it, also consistently make `piece_ids`/`attention_mask`
positional arguments, while making all other arguments kwarg-only.

## Checklist

- [x] I confirm that I have the right to submit this contribution under the project's MIT license.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

File: docs/source/deployment.rst [hunk 1]
```diff
@@ -36,16 +36,18 @@ inputs as a tuple. For example, we can trace a decoder as follows:
...
-   traced = torch.jit.trace(decoder, (X_example),))
+   mask_example = AttentionMask(torch.ones((4, 20), dtype=torch.bool, device=device))
+   traced = torch.jit.trace(decoder, (X_example, mask_example))
 
-As you can see, we are feeding the model with an all-zeros tensor during
-tracing. This is not really an issue, as long as the inputs allows the model to
-run normally, tracing can do its work.
+As you can see, we are feeding the model with an all-zeros piece identifier and
+all-ones mask tensors during tracing. This is not really an issue, as long as
+the inputs allows the model to run normally, tracing can do its work.
```
Comment (shadeMe) [inline docs/source/deployment.rst:50]:
```suggestion
As you can see, we are feeding the model with an all-zeros piece identifier tensor and
an all-ones mask tensor during tracing. This is not really an issue - as long as
the inputs allow the model to run normally, tracing can do its work.
```

**Review by shadeMe — APPROVED** [review]
LGTM!
